src/classes/objectProcess.ts

```diff
@@ -47,18 +47,46 @@ export class ObjectProcess {
 			AND TextTypeId = 67
 			AND TextEntry IS NOT NULL`;
 
-			// Execute the query to get the text entry
+			// Check if this copy differs from the last processed version of data we have
+			const storedTextEntryQuery = `
+			SELECT "textEntry"
+			FROM "text_entry_store"
+			WHERE "objectId" = ${this.objectId}
+			`;
+
+			// Execute the query to get the text entry from TMS 
 			const queryResult = await this.tmsCon.executeQuery(collectionPayloadQuery);
 			const recordSet = queryResult.recordset as CollectionPayloadTextEntry[];
 			const textEntryValue = recordSet[0]?.TextEntry;
 
 			if (Boolean(textEntryValue) === false) {
 				console.warn(`No "TextEntry" data available for Object ID "${this.objectId}"`, textEntryValue);
+				this.netxClient.release();
 				return resolve('');
 			}
-
 			const cpTextEntry = textEntryValue.replace(NEWLINE_RETURN_TAB_REGEX, '');
 
+			// Execute the query to get the stored text entry we have in the NetX intermediate database
+			const storedTextEntryQueryResult = await this.netxClient.query(storedTextEntryQuery);
+			const storedTextEntryRecordSet = storedTextEntryQueryResult?.rows;
+			const storedTextEntry = storedTextEntryRecordSet?.length ? storedTextEntryRecordSet[0].textEntry : ''
+
+			// If our stored version of the text entry matches the current one from TMS
+			// then there's been no new data for the record. We can move forward without
+			// doing any subsequent processing because the data is the same
+			if (storedTextEntry === cpTextEntry) {
+				this.netxClient.release();
+				return resolve('')
+			}
+
+			// At this point, the version of the TextEntry we have stored differs from
+			// what currently exists in TMS. Let's persist this new value into the TextEntryStore
+			console.info(`Stored version of TextEntry for Object ID ${this.objectId} is being updated`)
+			const { query: textEntryInsertQuery, values: textEntryInsertValues } = QueryHelpers.insertQueryGenerator(NetXTables.textEntryStore, {
+				objectId: this.objectId, textEntry: cpTextEntry, lastUpdatedAt: new Date()
+			})
+			await this.netxClient.query(textEntryInsertQuery, textEntryInsertValues);
+
```

Comment:
Code that checks if the version of the record in TMS matches what we have stored on our end - in  which case we return from the code since there's nothing to update as no change has been identified